src/customfielddialog.cpp

```diff
@@ -0,0 +1,311 @@
+                CustomDate->SetToolTip(Model_CustomField::getTooltip(field.PROPERTIES));
+                itemFlexGridSizer3->Add(CustomDate, g_flagsExpand);
+            }
+            break;
+        case Model_CustomField::TIME:
+            {
+                wxDateTime value;
+                if (fieldData->CONTENT.CmpNoCase("Now") == 0)
+                    value = wxDateTime::Now();
+                else if (!value.ParseTime(fieldData->CONTENT))
+                    value.ParseTime("00:00:00");
+                wxTimePickerCtrl* CustomDate = new wxTimePickerCtrl(this, controlID, value, wxDefaultPosition, wxDefaultSize, wxDP_DROPDOWN);
+                CustomDate->SetToolTip(Model_CustomField::getTooltip(field.PROPERTIES));
+                itemFlexGridSizer3->Add(CustomDate, g_flagsExpand);
+            }
+            break;
+        case Model_CustomField::SINGLECHOICHE:
+            {
+                wxArrayString Choiches = Model_CustomField::getChoiches(field.PROPERTIES);
+                Choiches.Sort();
+
+                wxChoice* CustomChoiche = new wxChoice(this, controlID, wxDefaultPosition, wxDefaultSize, Choiches);
+                CustomChoiche->SetToolTip(Model_CustomField::getTooltip(field.PROPERTIES));
+                itemFlexGridSizer3->Add(CustomChoiche, g_flagsExpand);
+
+                if (Choiches.size() == 0)
+                    CustomChoiche->Enable(false);
+
+                CustomChoiche->SetStringSelection(fieldData->CONTENT);
+            }
+            break;
+        default: break;
+        }
+    }
+
+    wxBoxSizer* itemBoxSizer22 = new wxBoxSizer(wxHORIZONTAL);
+    itemBoxSizer2->Add(itemBoxSizer22, wxSizerFlags(g_flags).Centre());
+
+    wxButton* itemButtonClose = new wxButton(this, wxID_CANCEL, _("Close"));
+    itemBoxSizer22->Add(itemButtonClose, g_flags);
+    itemButtonClose->SetToolTip(_("Close custom field window"));
+
+    wxBitmapButton* itemButtonAddEdit = new wxBitmapButton(this, wxID_EDIT
+        , mmBitmap(png::OPTIONS), wxDefaultPosition
+        , wxSize(itemButtonClose->GetSize().GetY(), itemButtonClose->GetSize().GetY()));
+    itemButtonAddEdit->SetToolTip(_("Add, edit and delete custom fields"));
+    itemBoxSizer22->Add(itemButtonAddEdit, g_flags);
+}
+
+void mmCustomFieldDialog::OnAddEdit(wxCommandEvent& /*event*/)
+{
+    mmCustomFieldListDialog dlg(this, m_RefType);
+    dlg.ShowModal();
+    m_RefreshRequested = dlg.GetRefreshRequested();
+}
+
+void mmCustomFieldDialog::OnCancel(wxCommandEvent& /*event*/)
+{
+    mmCustomFieldDialog::OnSave(false);
+}
+
+void mmCustomFieldDialog::OnQuit(wxCloseEvent& /*event*/)
+{
+    mmCustomFieldDialog::OnSave(false);
+}
+
+void mmCustomFieldDialog::OnMove(const wxPoint & RefPos, const wxSize & RefSize)
+{
+    this->Move(wxPoint(RefPos.x + RefSize.GetWidth(), RefPos.y));
+}
+
+void mmCustomFieldDialog::OnSave(const bool OpenStatus)
+{
+    if (m_RefreshRequested)
+    {
+        Destroy();
```

Comment:
Close it first then commit again.

2016-01-25 16:10 GMT+03:00 Gabriele-V <notifications@github.com>:

> In src/customfielddialog.cpp
> <https://github.com/moneymanagerex/moneymanagerex/pull/663#discussion_r50691022>
> :
>
> > +
> > +void mmCustomFieldDialog::OnQuit(wxCloseEvent& /*event*/)
> > +{
> > +    mmCustomFieldDialog::OnSave(false);
> > +}
> > +
> > +void mmCustomFieldDialog::OnMove(const wxPoint & RefPos, const wxSize & RefSize)
> > +{
> > +    this->Move(wxPoint(RefPos.x + RefSize.GetWidth(), RefPos.y));
> > +}
> > +
> > +void mmCustomFieldDialog::OnSave(const bool OpenStatus)
> > +{
> > +    if (m_RefreshRequested)
> > +    {
> > +        Destroy();
>
> PS: How to update pull request?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/moneymanagerex/moneymanagerex/pull/663/files#r50691022>
> .
>